+bids/layout.m

```diff
@@ -214,6 +214,15 @@
 
     file_list = return_file_list(modality, subject, schema);
 
+    % dependency previous file
+    dep_prev_files = struct( ...
+                            'index_group', 0, ...
+                            'group_base',  '', ...
+                            'group_len', 1, ...
+                            'index_data', 0, ...
+                            'data_base', '', ...
+                            'data_len', 1);
+
     for iFile = 1:size(file_list, 1)
 
       [subject, parsing] = bids.internal.append_to_layout(file_list{iFile}, ...
```

Comment:
It will be more efficient to not append to layout files with same base name but just copying the previous object.
This will allow to evade the recalculations of dependencies for the same file.